src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -108,4 +111,25 @@ private static CredentialsMatcher githubScanCredentialsMatcher() {
 
     private Connector() {}
 
+    /**
+     * Fail immediately and throw a customized exception.
+     */
+    public static final RateLimitHandler CUSTOMIZED = new RateLimitHandler() {
```

Comment:
@KostyaSha This implementation works fine. Please, find in my comment a console output. I cannot see the problem.